File: 백준/Bronze/5086. 배수와 약수/배수와 약수.py
import sys

def solve():
    for line in sys.stdin:
        if not line: return
        a, b = map(int, line.split())
        
        if a == 0 and b == 0: 
            break
        elif a % b == 0:
            print("multiple")
        elif b % a == 0:
            print("factor")
        else: 
            print("neither")        
        
if __name__ == "__main__":
    solve()

File: 백준/Bronze/5086. 배수와 약수/배수와 약수.cc
#include <iostream>
using namespace std;

int main(void) {
	ios_base::sync_with_stdio(false);
	cin.tie(nullptr);

	int a = 0, b = 0;
	while (cin >> a >> b) {
		if (a == 0 && b == 0) break;
		else if (a % b == 0) {
			cout << "multiple" << "\n";
		}
		else if (b % a == 0) {
			cout << "factor" << "\n";
		}
		else {
			cout << "neither" << "\n";
		}
	}

	return 0;
}